modules/siddhi-launcher/samples/sample-clients/wso2event-server/src/main/java/org/wso2/carbon/sample/server/DatabridgeTestServer.java

```diff
@@ -0,0 +1,154 @@
+* under the License.
+*/
+
+package org.wso2.carbon.sample.server;
+
+import org.apache.log4j.Logger;
+import org.wso2.carbon.databridge.commons.Credentials;
+import org.wso2.carbon.databridge.commons.Event;
+import org.wso2.carbon.databridge.commons.StreamDefinition;
+import org.wso2.carbon.databridge.commons.exception.MalformedStreamDefinitionException;
+import org.wso2.carbon.databridge.commons.utils.EventDefinitionConverterUtils;
+import org.wso2.carbon.databridge.core.AgentCallback;
+import org.wso2.carbon.databridge.core.DataBridge;
+import org.wso2.carbon.databridge.core.definitionstore.InMemoryStreamDefinitionStore;
+import org.wso2.carbon.databridge.core.exception.DataBridgeException;
+import org.wso2.carbon.databridge.core.exception.StreamDefinitionStoreException;
+import org.wso2.carbon.databridge.core.internal.authentication.AuthenticationHandler;
+import org.wso2.carbon.databridge.core.utils.AgentSession;
+import org.wso2.carbon.databridge.receiver.binary.conf.BinaryDataReceiverConfiguration;
+import org.wso2.carbon.databridge.receiver.binary.internal.BinaryDataReceiver;
+import org.wso2.carbon.databridge.receiver.thrift.ThriftDataReceiver;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Databridge Thrift Server which accepts Thrift/Binary events
+ */
+public class DatabridgeTestServer {
+    private static final String STREAM_NAME = "org.wso2.esb.MediatorStatistics";
+    private static final String VERSION = "1.0.0";
+    private static final Logger log = Logger.getLogger(DatabridgeTestServer.class);
+    private ThriftDataReceiver thriftDataReceiver;
+    BinaryDataReceiver binaryDataReceiver;
+    private InMemoryStreamDefinitionStore streamDefinitionStore;
+    private static final String STREAM_DEFN = "{" +
+            "  'name':'" + STREAM_NAME + "'," +
+            "  'version':'" + VERSION + "'," +
+            "  'nickName': 'Stock Quote Information'," +
+            "  'description': 'Some Desc'," +
+            "  'tags':['foo', 'bar']," +
+            "  'metaData':[" +
+            "          {'name':'ipAdd','type':'STRING'}" +
+            "  ]," +
+            "  'payloadData':[" +
+            "          {'name':'symbol','type':'STRING'}," +
+            "          {'name':'price','type':'DOUBLE'}," +
+            "          {'name':'volume','type':'INT'}," +
+            "          {'name':'max','type':'DOUBLE'}," +
+            "          {'name':'min','type':'Double'}" +
+            "  ]" +
+            "}";
+
+    public static void main(String args[]) throws DataBridgeException, InterruptedException,
+            StreamDefinitionStoreException, MalformedStreamDefinitionException {
+        DatabridgeTestServer databridgeTestServer = new DatabridgeTestServer();
+        databridgeTestServer.addStreamDefinition(STREAM_DEFN);
+        databridgeTestServer.start(args[0], Integer.parseInt(args[1]), args[2]);
+        Thread.sleep(100000000);
```

Comment:
can you explain he rationale behind this thread sleep?

Reply:
This was used in the product-sp as well. I used the same sample used in it.